src/main/java/com/slicingdice/jslicer/core/HandlerResponse.java

```diff
@@ -92,7 +96,8 @@ public boolean requestSuccessful() throws SlicingDiceException {
         try {
             data = new JSONObject(this.result);
         } catch (final JSONException exception) {
-            throw new InternalException("SlicingDice: Internal error.");
+            logger.severe(String.format("Couldn't parse JSON '%s'", this.result));
```

Comment:
This is an anti-pattern on exception handling, you should never log and throw on the same catch, or you log or you throw.